src/main/java/fi/hsl/pulsar/mqtt/MessageProcessor.java

```diff
@@ -36,7 +38,9 @@ public MessageProcessor(Config config, PulsarApplication pulsarApp, MqttConnecto
         this.pulsarApp = pulsarApp;
         this.producer = pulsarApp.getContext().getProducer();
         this.connector = connector;
+        this.lastMsgTimestamp = System.currentTimeMillis();
```

Comment:
I would remove this line and put to:
lastMsgReceivedTimeHealthy() {
    if (!this.lastMsgTimestamp) return true;
}
(so that it won't crash when constructor is called and first handleMessage call takes too long)